### PR TITLE
Delete `optuna.integration.ChainerPruningExtension` for migrating to optuna-integration package.

### PR DESCRIPTION
**Pull request by gen740**
## Motivation

`Optuna.integration` is a cluster of the integration modules that are independent of each other.
So splitting optuna-integraton into another module will reduce the cost of development.

I have made a prototype of [`optuna-integraton`](https://github.com/gen740/optuna-integration),
which only contains `optuna.integration.ChainerPruningExtension`.

In this PR, I delete the `optuna.integration.ChainerPruningExtension` for the `optuna-integration` module.

## Description of the changes

- Delete `optuna.integration.ChainerPruningExtension` implementation and tests.

### CODE REVIEW

File: .github/workflows/sphinx-build.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Sphinx
 on:
   push:
     branches:
-      - master
+      - optuna_integration_master
```
Comment (HideakiImamura) [inline .github/workflows/sphinx-build.yml:6]: This is unintended change?

File: .github/workflows/sphinx-build.yml [hunk 2]
```diff
@@ -39,6 +39,8 @@ jobs:
       run: |
         python -m pip install -U pip
         pip install --progress-bar off -U .[document]
+        # Integration module is needed for document of optuna-integration.
+        pip install git+https://github.com/gen740/optuna-integration.git
```
Comment (HideakiImamura) [inline .github/workflows/sphinx-build.yml:43]: The URL should be `git+https://github.com/optuna/optuna-integration.git`

File: .github/workflows/sphinx-build.yml [hunk 3]
```diff
@@ -84,6 +86,8 @@ jobs:
       run: |
         python -m pip install -U pip
         pip install --progress-bar off -U .[document]
+        # Integration module is needed for doctest.
+        pip install git+https://github.com/gen740/optuna-integration.git
```
Comment (HideakiImamura) [inline .github/workflows/sphinx-build.yml:90]: Ditto.

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR. I have several comments.

**Review by HideakiImamura — APPROVED** [review]
LGTM.

**Review by contramundum53 — APPROVED** [review]
LGTM!
Thank you for this nice work!

**Pull request merged**